### ISSUE TITLE
Add sphinx-autobuild to documentation build options

### ISSUE THREAD

**Issue by DON-BRAN** [zarr-developers/zarr-python#1244]
I believe it will be great to have [sphinx-autobuild](https://pypi.org/project/sphinx-autobuild/) added as a build option for the documentation. This can help avoid constant manual builds during development.

What are your thoughts please? 🤓

**Comment by joshmoore** [comment]
Are you suggesting adding an optional requirement? Strictly speaking, anyone could already install the package and use it locally now, right?

**Comment by DON-BRAN** [comment]
> Are you suggesting adding an optional requirement? Strictly speaking, anyone could already install the package and use it locally now, right?

Yes that's right anyone can install the package and use locally. Was just wondering if it could be shipped in the documentation build script to reduce the setup complexity for others

**Comment by joshmoore** [comment]
Guess I don't know exactly what that would look like, but happy to consider it.

**Comment by jhamman** [comment]
This works today using hatch:

```
hatch --env docs run serve
```
